pkg/utils/utils.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+// GetClusterDomainName returns cluster's domain name or error
+func GetClusterDomainName() (string, error) {
+	_, err := os.Stat("/etc/resolv.conf")
```

Comment:
I don't think you need to stat this before calling Open -- open will fail with an error if the file doesn't exst.

Reply:
removed